packages/components/src/components/input/input.stories.ts

```diff
@@ -69,731 +35,219 @@ export const Default = {
-            { type: 'attribute', name: 'clearable', value: true },
-            { type: 'attribute', name: 'value', value: 'value' },
-            { type: 'attribute', name: 'label', value: 'Label' },
-            { type: 'attribute', name: 'help-text', value: 'help-text' },
-            {
-              type: 'slot',
-              name: 'left',
-              value: '<sd-icon slot="left" library="global-resources" name="system/picture"></sd-icon>'
-            },
-            {
-              type: 'slot',
-              name: 'right',
-              value: '<sd-icon slot="right" library="global-resources" name="system/picture"></sd-icon>'
-            }
-          ],
-          args
-        })
-      )}
-    `;
-  }
+export const Required = {
+  render: () =>
+    html`<div class="w-[250px]">
+      <sd-input label="Label" required></sd-input>
+    </div>`
 };
 
 /**
- * Use the `form-control`, `form-control-label`, `form-control-input`, `form-control-help-text`, `base`, `border`, `input`, `left`, `clear-button`, and `right` part selectors to customize the input.
+ * The component gets `valid` state when the input is valid.
+ * Use the `style-on-valid` attribute to automatically indicate and show a valid state.
  */
 
-export const Parts = {
-  parameters: {
-    controls: {
-      exclude: ['label', 'left', 'right', 'clear-icon', 'help-text', 'clearable', 'value']
-    }
-  },
-  render: (args: any) => {
-    return generateTemplate({
-      axis: {
-        y: {
-          type: 'template',
-          name: 'sd-input::part(...){outline: solid 2px red}',
-          values: [
-            'form-control',
-            'form-control-label',
-            'form-control-input',
-            'form-control-help-text',
-            'base',
-            'border',
-            'input',
-            'left',
-            'clear-button',
-            'right'
-          ].map(part => {
-            return {
-              title: part,
-              value: `<style>#part-${part} sd-input::part(${part}){outline: solid 2px red}</style><div id="part-${part}">%TEMPLATE%</div>`
-            };
-          })
-        }
-      },
-      constants: [
-        { type: 'attribute', name: 'clearable', value: true },
-        { type: 'attribute', name: 'value', value: 'value' },
-        { type: 'attribute', name: 'label', value: 'Label' },
-        { type: 'attribute', name: 'help-text', value: 'help-text' },
-        {
-          type: 'slot',
-          name: 'left',
-          value: '<sd-icon slot="left" library="global-resources" name="system/picture"></sd-icon>'
-        },
-        {
-          type: 'slot',
-          name: 'right',
-          value: '<sd-icon slot="right" library="global-resources" name="system/picture"></sd-icon>'
-        }
-      ],
-      args
-    });
-  }
+export const Valid = {
+  render: () =>
+    html`<div class="w-[250px]">
+      <sd-input label="Label" style-on-valid></sd-input>
```

Comment:
Please fill "Input text here" here, so we can see the valid style out of the box.

![image](https://github.com/user-attachments/assets/824950af-bc73-44a9-a335-39052a28cae6)